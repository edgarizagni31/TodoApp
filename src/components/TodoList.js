import TodoListItem from './TodoListItem';

import './styles/TodoList.css';

const TodoList = ( { listTodos } ) => {
  return (
    <>
        {
            listTodos.map(( todo ) => (
                <TodoListItem key = { todo.id } todo = { todo } />
            ))
        }
    </>         
  );
};

export default TodoList;

