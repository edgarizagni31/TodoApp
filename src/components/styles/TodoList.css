.todo-list {
  background-color: hsl(235, 24%, 19%);
  border-radius: 4px;
  color: hsl(233,11%,84%);
  padding: 5px 10px;
  width: 100%;
}

.todo-list--empty  {
  padding: 5px;
}

.todo-list__items {
  width: 100%;
}

.todo-list__item {
  align-items: flex-end;
  border-bottom: 1px solid hsl(233, 11%, 84%);
  display: flex;
  height: 42px;
  padding-bottom: 10px;
  position: relative;
  width: 100%;
}

.todo-list__text {
  margin-left: 10px;
}

.todo-list__btn {
  border: 1px solid black;
  border-radius: 50%;
  background-color: #ffffff;
  height: 18px;
  outline: none;
  width: 18px;
}

.todo-list__icon {
  font-size: 18px;
  position: absolute;
  right: 0;
}

.complete {
  text-decoration: line-through;
  color: hsl(237, 14%, 26%);
}