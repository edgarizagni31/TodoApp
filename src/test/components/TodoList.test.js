import TodoList from '../../components/TodoList';
import { shallow } from 'enzyme'
import { demoTodos } from '../fixtures/demoTodo';

describe('test to TodoList', () => {
    const wrapper = shallow(<TodoList listTodos = {demoTodos}/>);

    test('should be displayed correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have demoTodos.length <TodoListItem />', () => {
        expect( wrapper.find('TodoListItem').length ).toBe(demoTodos.length);
    });
});
