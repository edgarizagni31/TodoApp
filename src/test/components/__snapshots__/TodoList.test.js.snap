// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to TodoList should be displayed correctly 1`] = `
<Fragment>
  <TodoListItem
    key="1"
    todo={
      Object {
        "desc": "Aprender mongo",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    key="2"
    todo={
      Object {
        "desc": "Aprender express",
        "done": false,
        "id": 2,
      }
    }
  />
</Fragment>
`;
